### PR TITLE
Construct a sparse array from a usual one

### PR DESCRIPTION
**Pull request by DanielRrr**
Needed for Nova as well

### CODE REVIEW

File: YatimaStdLib/SparseMatrix.lean [hunk 1]
```diff
@@ -109,6 +109,9 @@ instance : Add (SparseMatrix R) where
 
 abbrev SparseArray R := Std.RBMap Nat R compare
 
+def Array.toSparse (x : Array R) : SparseArray R :=
+  Std.RBMap.fromArray $ Array.zip (Array.iota x.size) (x : Array R)
```
Comment (arthurpaulino) [inline YatimaStdLib/SparseMatrix.lean:113]: Doesn't `iota` start from `1`?

**Review by arthurpaulino — COMMENTED** [review]
Do we need an extra check to only insert non-zero values? I feel like this beats the purpose of handling a sparse data structure

File: YatimaStdLib/SparseMatrix.lean (hunk 1 above)
Comment (DanielRrr) [reply YatimaStdLib/SparseMatrix.lean:113]: No, it starts from 0

**Comment by DanielRrr** [comment]
> Do we need an extra check to only insert non-zero values? I feel like this beats the purpose of handling a sparse data structure

What about the current version?

**Review by arthurpaulino — APPROVED** [review]

**Pull request merged**